node/subsystem/src/lib.rs

```diff
@@ -228,3 +228,33 @@ impl<C: SubsystemContext> Subsystem<C> for DummySubsystem {
 		}
 	}
 }
+
+/// A forward subsystem that implements [`Subsystem`].
+///
+/// It forwards all communication from the overseer to the internal message
+/// channel.
+///
+/// This susbsytem is useful for testing functionality that interacts with the overseer.
+pub struct ForwardSubsystem<Msg>(pub mpsc::Sender<Msg>);
```

Comment:
doesn't seem like it's used, are you planning to add something later?

Reply:
for tests we have `subsystem-test-helpers`, doesn't make sense to export test-only functionality it in `subsystem`

Reply:
I use it in cumulus and I thought that it will probably be useful in polkadot on the future.

`DummySubsystem` is in the same crate so I put it here, but I can move it.

Reply:
`DummySubsystem` is also used for doctests and currently in the real_overseer, but generally, we shouldn't publicly export test-only thing here.